src/relayFeeCalculator/chain-queries/baseQuery.ts

```diff
@@ -63,11 +60,23 @@ export default abstract class QueryBase implements QueryInterface {
    * Retrieves the current gas costs of performing a fillRelay contract at the referenced Spoke Pool
    * @returns The gas estimate for this function call (multplied with the optional buffer)
    */
-  async getGasCosts(): Promise<BigNumberish> {
-    const tx = await createUnsignedFillRelayTransaction(this.spokePool, this.usdcAddress, this.simulatedRelayerAddress);
+  async getGasCosts(
+    deposit: Deposit,
+    amountToRelay: BigNumberish,
+    relayAddress = DEFAULT_SIMULATED_RELAYER_ADDRESS,
+    _relayerBalanceForToken?: BigNumberish
```

Comment:
For the `_relayerBalanceForToken`? Yes, this is so that the `relayer` can pass its erc20 balance to the SDK function without having an additional RPC call to resolve the balance.

Reply:
This has now been removed. Closing this comment